redis.go

```diff
@@ -96,9 +96,17 @@ func (c *baseClient) WrapProcess(fn func(oldProcess func(cmd Cmder) error) func(
 
 func (c *baseClient) defaultProcess(cmd Cmder) error {
 	for i := 0; i <= c.opt.MaxRetries; i++ {
+		if i > 0 {
+			time.Sleep(internal.RetryBackoff(i, c.opt.MaxBackoffRetryTimeout))
+		}
+
 		cn, _, err := c.conn()
 		if err != nil {
 			cmd.setErr(err)
+
+			if err != nil && internal.IsRetryableError(err) {
```

Comment:
Should the ```err != nil &&``` check be removed from https://github.com/go-redis/redis/pull/570/files#diff-bbc64daf788a83d9a6daf7056d49d2a8R117 as well? Also, I noticed that ```err = readReply(cn)``` (https://github.com/go-redis/redis/pull/570/files#diff-bbc64daf788a83d9a6daf7056d49d2a8R124) is missing a corresponding ```cmd.setErr(err)```. Is that necessary to run that function when readReply returns error?